src/lib/components/SettingsDialog/SettingsDialog.svelte

```diff
@@ -70,6 +70,10 @@
 		meltUiOpen.set(true);
 		activeTab = 'workspace';
 	};
+
+	$effect(() => {
+		document.body.setAttribute('data-theme', userData.theme ?? 'dark');
+	});
```

Comment:
The settings dialog component should ideally be UI-only, without side effects / logic like this. Can we move this to `routes/[id]/+page.svelte` instead?